src/Gateways/PagSeguro/Holder.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace BeerAndCodeTeam\PaymentGateway\Gateways\PagSeguro;
+
+class Holder
+{
+    private string $name;
+    private string $taxId;
+    private string $email;
+    private Address $address;
+
+    public function __contruct()
+    {
+        $this->address = new Address;
+    }
+
+    public function setName(string $name)
+    {
+        if (strlen($name) < 1  || strlen($name) > 30) {
+            throw new \Exception("Error:  the name most be 1-30 characters. It has " . strlen($name));
+        }
+        $this->name = $name;
+    }
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    public function setTaxId(string $taxId)
+    {
+        if (strlen($taxId) != 11 && strlen($taxId) != 14) {
+            throw new \Exception("Error: the taxId most be 11 or 14 characters. It has " . strlen($taxId));
+        }
+        $this->taxId = $taxId;
+    }
+
+    public function getTaxId()
+    {
+        return $this->taxId;
+    }
+
+    public function setEmail(string $email)
+    {
+        if (strlen($email) < 10 || strlen($email) > 255 || !filter_var($email, FILTER_VALIDATE_EMAIL)) {
```

Comment:
a função **filter_var** já não verifica o email?.. ele precisa ser de um tamanho específico?

outro ponto, todas as condições usam **| |**, ou seja, se o tamanho da string **$email** for maior que 10, não cairá na exceção, mesmo que tenha mais de 255 caracteres e não tenha passado na validação de email.

Reply:
Pq segundo a documentação o email deve ter no mínimo 10 e no máximo 255 caracteres a idéia é barrar a requisição caso não atenda

Reply:
Acho que eu olhei a condição para passar e não para invalidar heheh